com.woltlab.wcf/templates/error.tpl

```diff
@@ -0,0 +1,29 @@
+{if ENABLE_DEBUG_MODE}
+{if $exception !== null}
+<!--
+{* A comment may not contain double dashes. *}
+{@'--'|str_replace:'- -':$exception}
+-->
+{/if}
+{/if}
```

Comment:
This might mess up the encoding of characters, because there is potentially UTF-8 content before the document starts. At that point you dive deep into “garbage in – garbage out” territory, because you’re violating the specs.

You could move it right after the include of the `header` template and add a unique string that serves as a marker like this:
https://github.com/WoltLab/WCF/blob/6b9ea31ed7959bad65793f5c28339f509005ef86/com.woltlab.wcf/templates/pageFooter.tpl#L2-L18